runtime/reconcilers/canvas_test.go

```diff
@@ -192,6 +193,13 @@ first_month_of_year: 1
 	// Reconcile and verify that the metrics view gets updated
 	testruntime.ReconcileParserAndWait(t, rt, id)
 
+	// Add polling logic to wait for expected state
+	require.Eventually(t, func() bool {
+		mv2 = testruntime.GetResource(t, rt, id, runtime.ResourceKindMetricsView, "mv2")
+		return mv2.GetMetricsView().State.ValidSpec != nil &&
+			mv2.GetMetricsView().State.ValidSpec.FirstDayOfWeek == 1
+	}, 5*time.Second, 100*time.Millisecond, "metrics view spec not updated in time")
```

Comment:
Not sure, but I don't think you need to poll. The `ReconcileParserAndWait` call will wait until nothing is reconciling.

I think the problem is just that the order iteration is non-deterministic, so the check on line `212` sometimes have the names the other way around.

Reply:
Ahh yep